### ISSUE TITLE
Core dump on Cygwin when $PWD is a Windows path

### ISSUE THREAD

**Issue by thorstenkampe** [fish-shell/fish-shell#7636]
```
> fish --version
fish, version 3.1.2

> echo $version
3.1.2

> echo $PWD
F:\cygwin\home\thorsten

> c
assertion "!wd.empty() && wd.front() == sep && wd.back() == sep && "Invalid working directory, it must start and end with /"" failed: file "/home/ASchulma/dev/cygwin/fish/fish-3.1.2-2.x86_64/src/fish-3.1.2/src/wutil.cpp", line 482, function: wcstring path_normalize_for_cd(const wcstring&, const wcstring&)
      0 [fish] fish 9070 cygwin_exception::open_stackdumpfile: Dumping stack trace to fish.exe.stackdump
```

Running fish on Cygwin from a shell which sets $PWD to a Windows path (PowerShell with Starship prompt) results in a core dump (see above).

**Comment by faho** [comment]
Some more information would be useful. Like the coredump itself, what exactly you mean by "windows path", and how you get one to be $PWD.

**Comment by thorstenkampe** [comment]
Sorry, the initial description was sent prematurely (keyboard error)

**Comment by faho** [comment]
Okay, so the issue here is that your path isn't like a unix-path starting from "/" on.

This asserts in normalizing the path, and if it didn't it would do something entirely non-sensical - for instance it doesn't handle `\\`, because that's not a unix thing.

We should probably just get the path again via getcwd if it doesn't start with a `/`.

**Comment by faho** [comment]
Should be fixed with 8b13383 - when fish inherits a path like that it doesn't trust it and uses getcwd() instead.

Either wait for 3.2, build fish yourself or get the cygwin people to backport it.
